c7n/filters/iamaccess.py

```diff
@@ -126,9 +126,14 @@ def handle_principal(self, s):
         if not isinstance(principals, dict):
             principals = {'AWS': principals}
 
+        principals_set = set(principals)
```

Comment:
nit, for clarity it would be good to name this principal_types .. set could be mistaken for the actual principal values.